### DISCUSSION TITLE
Sonoff Dual R3 stuck after Tasmota firmware flash

### DISCUSSION THREAD

**Discussion by fft2865** [arendst/Tasmota#19812] [Support request] [answered]
Before I outline my issue a few words about my level of expierence. I'm well familiar with all kinds of electronics but never dived deep into the world of microcontrollers. I know how to flash Tasmota on Sonoffs and did it for some 50+ with different types. My hardware never let me down, so I was really surprised that I could not get the Dual R3 up and running after the flash. 

I bought four Sonoff Dual R3 V2.0 2 weeks ago which I tried to equip with Tasmota. Whatever tool I used to write the firmware and even with different Tasmota versions, with and without factory-bootloader, latest version or first version that supports the Dual R3, I get the same result: the blue light stays dimmed after the power cyle, the device does not boot. I went through all available docs/discussion here and finally tried the latest esp-tool with 

esptool.py --chip esp32 --port COM5 --baud 921600 --before default_reset --after hard_reset write_flash -z --flash_mode dout --flash_freq 40m --flash_size detect 0x00000 tasmota32.factory.bin

which sounded very promising but brought the same result. 

My assumption was, that something is wrong with my hardware, so I read the firmware from one new Dual R3 and wrote it back to my test device, which worked seamlessly. After the power cycle the R3 immeadiatly came up. Conclusion: the hardware works, but the test device can not handle any of the the current tasmota firmware.

Interesting finding with the device I brought back by cloning the firmware from an untouched device: it has the same ID in the sonoff app as the device donating the firmware. Seems each device gets an individual firmware during production with the ID coded in the firmware.

Any ideas what else I can do?

**Comment by sfromis** [comment]
It was not useful to try without it being with `.factory` firmware.

You do not say whether you watched serial while booting, which may provide info. However, this should not be done while doing what commonly is needed to make ESP32-based devices work; to boot with mains power instead of the flashing setup.

**Comment by Jason2866** [comment]
Don't provide flash mode and size info when flashing a factory firmware file. It can damage the factory file depending on which version of esptool.py you use. Less is better ;-)
```
esptool.py --chip esp32 --port COM5 --baud 460800 write_flash 0x00000 tasmota32.factory.bin
```

**Reply by fft2865** [reply]
Thanks a mil for the recommendations. Unfortunately it did work. Here is what I get when I scan my ports:

>> esptool.exe flash_id
esptool.py v4.6.2
Found 5 serial ports
Serial port COM7
Connecting.......
Detecting chip type... Unsupported detection protocol, switching and trying again...
Connecting...
Detecting chip type... ESP32
Chip is ESP32-D0WD-V3 (revision v3.0)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: 3c:e9:0e:86:94:44
Uploading stub...
Running stub...
Stub running...
Manufacturer: 5e
Device: 4016
Detected flash size: 4MB
Hard resetting via RTS pin...

Now writing the factory.bin with:
>>esptool.py --chip esp32 --port COM7 --baud 115000 write_flash 0x00000 tasmota32-DE.factory.bin
esptool.py v3.3.3
Serial port COM7
Connecting...
Chip is ESP32-D0WD-V3 (revision v3.0)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: 3c:e9:0e:86:94:44
Stub is already running. No upload is necessary.
Configuring flash size...
Flash will be erased from 0x00000000 to 0x002b3fff...
Compressed 2834384 bytes to 1819278...
Wrote 2834384 bytes (1819278 compressed) at 0x00000000 in 165.2 seconds (effective 137.3 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...

[The high speed of 460800  leads to errors, half of that works]

I see the same problem, blue light stays dimmed, no boot (I tried the original tasmota32.factory.bin as well as the tasmota32-DE.factory.bin). This behavior is independent of the power supply, I tried both direct 3.3 V inject as well as 220V AC via the screw terminal (of course after disconnecting the flash cables). To verify that the R3 is still worky I finally wrote the initial firmware back, which leads to immediate boot after power cycling even with the power support provided from the flash cables.

I have attached 4 pics of the device.

![DualR3-V2 0-1](https://github.com/arendst/Tasmota/assets/148117251/0df96bc8-89ee-4a4e-9c60-f99e4dc51f51)
![DualR3-V2 0-2](https://github.com/arendst/Tasmota/assets/148117251/91104d9b-edd8-4338-8c29-3e1d98ee6585)
![DualR3-V2 0-3](https://github.com/arendst/Tasmota/assets/148117251/71d533dc-d2d8-4425-8293-94f684ac92e7)
![DualR3-V2 0-4](https://github.com/arendst/Tasmota/assets/148117251/95d82faf-cdc5-4c30-be5a-674680f8e9dd)

**Comment by arendst** [comment, marked as answer]
Why do you expect the blue led to light? As long as the default template is loaded no light will shine.

You'll need to monitor serial for some initial data (it won't start completely due to insufficient power) or connect AC and search for a wifi AP with tasmota in the name.

**Reply by fft2865** [reply]
You are right, my mistake. I expected similar behavior to the original firmware that shows some flashing activity during boot. 
Problem solved, the R3 comes up with the tasmota32.factory.bin firmware, configuration template works too. 
Thanks a mil for your help.
